src/main/kotlin/no/nav/familie/ef/sak/selvstendig/NæringsinntektKontrollService.kt

```diff
@@ -0,0 +1,132 @@
+package no.nav.familie.ef.sak.selvstendig
+
+import no.nav.familie.ef.sak.amelding.InntektService
+import no.nav.familie.ef.sak.behandling.BehandlingService
+import no.nav.familie.ef.sak.behandling.domain.Behandling
+import no.nav.familie.ef.sak.fagsak.FagsakService
+import no.nav.familie.ef.sak.felles.util.kuttPeriodeTilGittÅr
+import no.nav.familie.ef.sak.oppgave.OppgaveService
+import no.nav.familie.ef.sak.oppgave.OppgaveUtil
+import no.nav.familie.ef.sak.sigrun.SigrunService
+import no.nav.familie.ef.sak.vedtak.VedtakService
+import no.nav.familie.kontrakter.felles.Behandlingstema
+import no.nav.familie.kontrakter.felles.Månedsperiode
+import no.nav.familie.kontrakter.felles.Tema
+import no.nav.familie.kontrakter.felles.ef.StønadType
+import no.nav.familie.kontrakter.felles.oppgave.FinnOppgaveRequest
+import no.nav.familie.kontrakter.felles.oppgave.IdentGruppe
+import no.nav.familie.kontrakter.felles.oppgave.Oppgave
+import no.nav.familie.kontrakter.felles.oppgave.Oppgavetype
+import no.nav.familie.leader.LeaderClient
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.time.YearMonth
+import java.util.UUID
+
+@Service
+class NæringsinntektKontrollService(
+    val oppgaveService: OppgaveService,
+    val fagsakService: FagsakService,
+    val behandlingService: BehandlingService,
+    val vedtakService: VedtakService,
+    val sigrunService: SigrunService,
+    val inntektService: InntektService,
+) {
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+    private val årstallIFjor = YearMonth.now().minusYears(1).year
+
+    fun sjekkNæringsinntektMotForventetInntekt() {
+        if (LeaderClient.isLeader() == true) {
+            val oppgaver = hentOppgaverForSelvstendigeTilInntektskontroll()
+
+            oppgaver.forEach { oppgave ->
+                val personIdent =
+                    oppgave.identer?.firstOrNull { it.gruppe == IdentGruppe.FOLKEREGISTERIDENT }?.ident
+                        ?: oppgave.personident ?: throw Exception("Fant ikke registrert ident på oppgave ${oppgave.id}")
+
+                secureLogger.info("Kontrollerer person med ident: $personIdent")
+
+                val fagsakOS = fagsakService.finnFagsaker(setOf(personIdent)).filter { it.stønadstype == StønadType.OVERGANGSSTØNAD }.firstOrNull() ?: throw RuntimeException("Fant ikke fagsak for overgangsstønad for person: $personIdent")
+                val behandling = behandlingService.finnSisteIverksatteBehandling(fagsakOS.id) ?: throw RuntimeException("Fant ikke behandling for fagsakId: ${fagsakOS.id}")
+                val antallMånederMedVedtakForFjoråret = antallMånederMedVedtakForFjoråret(behandling)
+
+                val næringsinntekt = hentFjoråretsNæringsinntekt(fagsakOS.fagsakPersonId)
+
+                if (antallMånederMedVedtakForFjoråret > 3 && næringsinntekt > 50_000) {
```

Comment:
antar dette er en slags grense - kan den gis et logisk navn?

Reply:
Ja, hvis næringsinntekt er 50 000 eller lavere så må det bes om regnskap for å se om bruker oppfyller aktivitetskravet. Mulig det kan hete noe ala `INNTEKTSGRENSE_FOR_KONTROLL_AV_AKTIVITET` ?